File: app/src/main/java/org/jellyfin/androidtv/data/model/FilterOptions.java
package org.jellyfin.androidtv.data.model;

import org.jellyfin.apiclient.model.querying.ItemFilter;

import java.util.ArrayList;
import java.util.List;

public class FilterOptions {
    private boolean favoriteOnly;
    private boolean unwatchedOnly;

    public void setFavoriteOnly(boolean value) {
        favoriteOnly = value;
    }

    public void setUnwatchedOnly(boolean value) {
        unwatchedOnly = value;
    }

    public boolean isUnwatchedOnly() {
        return unwatchedOnly;
    }

    public boolean isFavoriteOnly() {
        return favoriteOnly;
    }

    public ItemFilter[] getFilters() {
        if (!unwatchedOnly && !favoriteOnly) return null;

        List<ItemFilter> filters = new ArrayList<>();
        if (favoriteOnly) filters.add(ItemFilter.IsFavorite);
        if (unwatchedOnly) filters.add(ItemFilter.IsUnplayed);

        return filters.toArray(new ItemFilter[0]);
    }
}

File: app/src/main/java/org/jellyfin/androidtv/data/model/FilterOptions.kt
package org.jellyfin.androidtv.data.model

import org.jellyfin.sdk.model.api.ItemFilter


class FilterOptions {
	var isFavoriteOnly = false
	var isUnwatchedOnly = false
	var selectedGenre: String? = null

	val filters: Set<ItemFilter>
		get() = buildSet {
			if (isFavoriteOnly) add(ItemFilter.IS_FAVORITE)
			if (isUnwatchedOnly) add(ItemFilter.IS_UNPLAYED)
		}

	val genres: Set<String>?
		get() = selectedGenre
			?.takeIf { it.isNotBlank() }
			?.let(::setOf)

	fun hasGenre() = !selectedGenre.isNullOrBlank()

	fun hasFilters() = isFavoriteOnly || isUnwatchedOnly || hasGenre()
}
